rotkehlchen/exchanges/bybit.py

```diff
@@ -0,0 +1,514 @@
+
+        return new_trades, (start_ts, end_ts)
+
+    def validate_api_key(self) -> tuple[bool, str]:
+        """Validates that the Bybit API key is good for usage in rotki"""
+        try:
+            self._api_query(path='user/query-api')
+        except RemoteError as e:
+            return False, str(e)
+        else:
+            return True, ''
+
+    def query_balances(self, **kwargs: Any) -> ExchangeQueryBalances:
+        """
+        Query balances at bybit.
+
+        Known limitations:
+        - It can't query balance deposited in bots.The API doesn't provide this information
+        """
+        self.first_connection()
+        assets_balance: defaultdict[AssetWithOracles, Balance] = defaultdict(Balance)
+
+        try:
+            response = self._api_query(
+                path='account/wallet-balance',
+                options={
+                    'accountType': 'UNIFIED' if self.is_unified_account else 'SPOT',
+                },
+            )
+        except RemoteError as e:
+            msg = f'Bybit request failed. Could not reach the exchange due to {e!s}'
+            log.error(msg)
+            return None, msg
+
+        log.debug(f'Bybit account response: {response}')
+        for account in response['list']:
+            for coin_data in account['coin']:
+                try:
+                    asset = asset_from_bybit(coin_data['coin'])
+                    amount = deserialize_fval(coin_data['walletBalance'], name=f'Bybit wallet balance for {asset}', location='bybit')  # noqa: E501
+                    if coin_data['usdValue'] != '':
+                        usd_value = deserialize_fval(coin_data['usdValue'], name=f'Bybit usd value for {asset}', location='bybit')  # we don't need to calculate it since it is provided by bybit  # noqa: E501
+                    else:
+                        usd_price = Inquirer().find_usd_price(asset=asset)
+                        usd_value = usd_price * amount
+                except UnknownAsset as e:
+                    self.msg_aggregator.add_warning(
+                        f'Found Bybit balance result with unknown asset '
+                        f'{e.identifier}. Ignoring it.',
+                    )
+                    continue
+                except (DeserializationError, KeyError) as e:
+                    msg = str(e)
+                    if isinstance(e, KeyError):
+                        msg = f'Missing key entry for {msg}.'
+                    return None, f'Error processing Bybit balance entry {coin_data}. {msg}'
+
+                assets_balance[asset] += Balance(amount=amount, usd_value=usd_value)
+
+        return assets_balance, ''
+
+    def _query_deposits(self, options: dict[str, Any]) -> list[AssetMovement]:
+        """
+        Process deposits from bybit. If any asset is unknown or we fail to deserialize an entry
+        the error is logged and we skip the entry.
+        """
+        log.debug(f'querying online deposits for {options}')
```

Comment:
Same question here. Won't you see the api query inside? Also why no bybit specific text?